joomla/joomla-cms
# [com_contact] add searchtools to modal layout and solves bugs

#### Description

This PR adds searchtools to com_contact modal layout contacts view.

Also, in normal layout:
- correct bug: by default the filter was open with no filters selected.
- correct bug: when selecting the category filter or the tag filter the searchtools hides.
- improves responsive visual (mobile/tablet/desktop)
- solves other minor graphical issues.
- minor code style improvements
###### Before PR

![image](https://cloud.githubusercontent.com/assets/9630530/13271287/f45afe6e-da8b-11e5-98ff-ccb50aab553e.png)
###### After PR

![image](https://cloud.githubusercontent.com/assets/9630530/13271015/ebf1db64-da89-11e5-8d4b-a6762f114898.png)
###### Before PR (modal)

![image](https://cloud.githubusercontent.com/assets/9630530/13271253/c6e04480-da8b-11e5-9058-3d21521b040f.png)
###### After PR  (modal)

![image](https://cloud.githubusercontent.com/assets/9630530/13271069/6ba2d408-da8a-11e5-8473-7a6ac41f2b26.png)
#### How to test

Normal view:
1. Install latest staging and apply this patch
2. Go to Components -> Contacts
3. Check if search, filters, ordering and pagination are working properly
4. Test on mobile and tablet (ex: reduce browser width)

Modal view:
1. Enable language associations.
2. Create some contacts in several languages.
3. Edit one of those contacts and go to "Associations" tab and click on "Select" on any language.
4. Check if search, filters, ordering and pagination are working properly in modal.
5. Test on mobile and tablet (ex: reduce browser width)
